tools/hsmtool.c

```diff
@@ -595,31 +596,38 @@ static int dumponchaindescriptors(const char *hsm_secret_path, const char *old_p
 	} while (bip32_key_from_seed(bip32_seed, sizeof(bip32_seed),
 				     version, 0, &master_extkey) != WALLY_OK);
 
-	if (bip32_key_to_base58(&master_extkey, BIP32_FLAG_KEY_PUBLIC, &enc_xpub) != WALLY_OK)
-		errx(ERROR_LIBWALLY, "Can't encode xpub");
+	if (show_secrets) {
+		if (bip32_key_to_base58(&master_extkey, BIP32_FLAG_KEY_PRIVATE,
+					&enc_xkey) != WALLY_OK)
+			errx(ERROR_LIBWALLY, "Can't encode xpriv");
+	} else {
+		if (bip32_key_to_base58(&master_extkey, BIP32_FLAG_KEY_PUBLIC,
+					&enc_xkey) != WALLY_OK)
+			errx(ERROR_LIBWALLY, "Can't encode xpub");
+	}
```

Comment:
```suggestion
	if (show_secrets) {
		if (bip32_key_to_base58(&master_extkey, BIP32_FLAG_KEY_PRIVATE,
					&enc_xkey) != WALLY_OK)
			errx(ERROR_LIBWALLY, "Can't encode xpriv");
	} else {
		if (bip32_key_to_base58(&master_extkey, BIP32_FLAG_KEY_PUBLIC,
					&enc_xkey) != WALLY_OK)
			errx(ERROR_LIBWALLY, "Can't encode xpub");
	}
```
```suggestion
	if (show_secrets)
		if (bip32_key_to_base58(&master_extkey, BIP32_FLAG_KEY_PRIVATE,
					&enc_xkey) != WALLY_OK)
			errx(ERROR_LIBWALLY, "Can't encode xpriv");
	else
		if (bip32_key_to_base58(&master_extkey, BIP32_FLAG_KEY_PUBLIC,
					&enc_xkey) != WALLY_OK)
			errx(ERROR_LIBWALLY, "Can't encode xpub");
```

We do not use `{}` for one line if

Reply:
This is actually not possible because of this error
```
tools/hsmtool.c:599:12: error: suggest explicit braces to avoid ambiguous 'else' [-Werror=dangling-else]
```

Reply:
uh nice!